PolyclinicLab/Polyclinic.Api.Host/Controllers/AppointmentsController.cs

```diff
@@ -0,0 +1,204 @@
+            }
+
+            if (dto.DoctorId <= 0)
+            {
+                logger.LogWarning("Create appointment called with invalid doctor ID: {DoctorId}", dto.DoctorId);
+                return BadRequest("Doctor ID must be greater than 0");
+            }
+
+            if (dto.PatientId <= 0)
+            {
+                logger.LogWarning("Create appointment called with invalid patient ID: {PatientId}", dto.PatientId);
+                return BadRequest("Patient ID must be greater than 0");
+            }
+
+            var createdAppointment = service.Create(dto);
+            logger.LogInformation("Successfully created appointment {Id} for doctor {DoctorId} and patient {PatientId}", 
+                createdAppointment.Id, dto.DoctorId, dto.PatientId);
+            
+            return CreatedAtAction(nameof(Get), new { id = createdAppointment.Id }, createdAppointment);
+        }
+        catch (ArgumentException ex)
+        {
+            logger.LogWarning(ex, "Validation error while creating appointment");
+            return BadRequest(ex.Message);
+        }
+        catch (Exception ex)
+        {
+            logger.LogError(ex, "Error occurred while creating appointment");
+            return StatusCode(500, "An error occurred while creating the appointment");
+        }
+    }
+
+    /// <summary>
+    /// Updates an existing appointment by its ID.
+    /// </summary>
+    /// <param name="id">The ID of the appointment to update.</param>
+    /// <param name="dto">The updated appointment data.</param>
+    [HttpPut("{id}")]
+    public ActionResult<AppointmentDto> Update(int id, [FromBody] CreateUpdateAppointmentDto dto)
+    {
+        try
+        {
+            // Validate input data
+            if (id <= 0)
+            {
+                logger.LogWarning("Update appointment called with invalid ID: {Id}", id);
+                return BadRequest("Appointment ID must be greater than 0");
+            }
+
+            if (dto == null)
+            {
+                logger.LogWarning("Update appointment called with null DTO for ID: {Id}", id);
+                return BadRequest("Appointment data is required");
+            }
+
+            if (dto.Room <= 0)
+            {
+                logger.LogWarning("Update appointment called with invalid room: {Room} for ID: {Id}", dto.Room, id);
+                return BadRequest("Room number must be greater than 0");
+            }
+
+            if (dto.DoctorId <= 0)
+            {
+                logger.LogWarning("Update appointment called with invalid doctor ID: {DoctorId} for ID: {Id}", dto.DoctorId, id);
+                return BadRequest("Doctor ID must be greater than 0");
+            }
+
+            if (dto.PatientId <= 0)
+            {
+                logger.LogWarning("Update appointment called with invalid patient ID: {PatientId} for ID: {Id}", dto.PatientId, id);
+                return BadRequest("Patient ID must be greater than 0");
+            }
```

Comment:
Удобнее в DTO добавить атрибуты валидации. К примеру:

Пример:
```c#
public record CreateUpdateAppointmentDto(
    DateTime Date,
    [Range(1, int.MaxValue, ErrorMessage = "Room number must be greater than 0")]
    int Room,
    bool IsRepeated,
    [Range(1, int.MaxValue, ErrorMessage = "Doctor ID must be greater than 0")]
    int DoctorId,
    [Range(1, int.MaxValue, ErrorMessage = "Patient ID must be greater than 0")]
    int PatientId
);
```
Таким образом валидация в контроллере была бы очень проста:


```suggestion
            if (!ModelState.IsValid)
            {
                logger.LogWarning("Create appointment validation failed: {Errors}",
                    string.Join("; ", ModelState.Values
                        .SelectMany(v => v.Errors)
                        .Select(e => e.ErrorMessage)));
                return BadRequest(ModelState);
            }
```